syntax = "proto3";

service Booking {
    rpc GetAllBookings(BookingEmpty) returns (Bookings) {}
    rpc GetUserBookings(UserId) returns (Bookings) {}
    rpc AddBookingOfUser(BookingItem) returns (Message) {}
    rpc GetScheduleByDateB(BDate) returns (BSchedule) {}
    rpc GetMovieScheduleB(BMovieID) returns (BMovieSchedule) {}
}

message BookingEmpty {
}

message Bookings {
    repeated BookingItem bookings =  1;
}

message UserId {
    string id = 1;
}

message BookingItem {
    string user = 1;
    string date = 2;
    string movie = 3;
}

message Message {
    string body = 1;
}

message BDate {
    string date = 1;
}

message BSchedule {
    string date = 1;
    repeated string movies = 2;
}

message BMovieID {
    string id = 1;
}

message BMovieSchedule {
    string movie = 1;
    repeated string dates =2;
}